src/debugger/Debugger.cpp

```diff
@@ -262,10 +270,27 @@ bool Debugger::doEval(ExecutionState* state, ByteCodeBlock* byteCodeBlock, uint8
     type = ESCARGOT_MESSAGE_EVAL_RESULT_8BIT;
     m_stopState = ESCARGOT_DEBUGGER_IN_EVAL_MODE;
 
+
     try {
         Value asValue(str);
-        Value result(state->context()->globalObject()->evalLocal(*state, asValue, Value(Value::Undefined), byteCodeBlock->m_codeBlock->asInterpretedCodeBlock(), true));
-        str = result.toStringWithoutException(*state);
+        if (index != 0) {
```

Comment:
I think we only have lex env index, but we also need a state depth as well (see other functions for the correct name). First we go up in states, then in lex envs.